packages/api/src/hooks/p2p/useAdvertInfo.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import useQuery from '../../useQuery';
+
+/**
+ * This custom hook returns the advert info for a specific advert by calling 'p2p_advert_info' endpoint
+ */
+const useAdvertInfo = (
+    payload: NonNullable<Parameters<typeof useQuery<'p2p_advert_info'>>[1]>['payload'],
```

Comment:
Do we need to pass the whole payload to it? 

what about this? 🤔 
```suggestion
    id?: number, use_client_limits?: [not sure about the type]
```

Reply:
should be a number, either 1 or 0

Reply:
@niloofar-deriv, so far the convention has been implementing them with `payload`, and the types are automatically inferred with this approach. I can change them if you guys think I should though 😁

Reply:
Also, `id` is a `string` Should I change it @niloofar-deriv @ameerul-deriv ?

Reply:
IMO i think its fine to have it as payload, in the future, if for whatever reason they want to add more things that we can send to the api endpoint, we have to amend this whole hook. The payload should be more than sufficient to handle this

Reply:
what do you think @niloofar-deriv

Reply:
Yeah @ameerul-deriv agree, I forgot the hooks are general and we should not limit them like this